### DISCUSSION TITLE
Exploring the New Actions Performance Metrics Now in Public Preview 👀⏱️

### DISCUSSION THREAD

**Discussion by samus-aran** [community/community#142513] [Actions]
We are excited to introduce a new observability UI that provides insights into the performance of your workflows and jobs for your organizations or repositories, now available in public preview! Performance metrics are designed to help you monitor, analyze, and optimize your Actions workflows and job execution. With these insights, you can answer common questions like:

- **How long does it take for my workflows or jobs to complete?** Gain visibility into the end-to-end execution times of your workflows and jobs, helping you identify bottlenecks and areas for improvement.
- **How long are my workflows or jobs waiting to run?** View queue times to identify bottlenecks in job scheduling, enabling you to make adjustments that reduce idle time and improve throughput.
- **Which of my workflows or jobs are consistently failing?** Spot frequent failures in your workflows, pinpoint root causes, and take preemptive actions to improve reliability and performance.
- **Where are my longest-running workflows or jobs originating from?**
Get insights on job origin points, allowing you to address lengthy processes directly and refine areas with the most significant impact.

![alt_text]( https://github.com/user-attachments/assets/a5a5523f-28d3-4415-8ae9-bc9653f29dc1 "image_tooltip")

### Getting Started ➡️ 

To start leveraging these insights, navigate to your organization or repository homepage, select **“Insights” **near the top of the page, and then choose **“Actions Performance Metrics”** on the left side of the page. Once you’re there, you’ll have public preview access to detailed metrics on your workflows and jobs, offering a comprehensive view of performance at every stage.

### GitHub Actions Metrics for Free, Pro, and Team plans 🆓 

We are also pleased to announce that with today's release, GitHub Actions Metrics are now available to Free, Pro, and Team plans. Previously, this feature was [only available to those on the GitHub Enterprise Cloud plan](https://github.com/orgs/community/discussions/132817). 

To learn more about GitHub Actions Metrics, check out our [public documentation](https://docs.github.com/en/actions/administering-github-actions/viewing-github-actions-metrics)

### Share your Feedback 💬 

We’re committed to making Actions Performance metrics as helpful and actionable as possible, and hearing directly from you will guide future improvements. As you explore the new feature, keep these questions in mind to help guide your experience:

- **Metrics or Data Points:** Are there additional metrics or data points you’d like to see included? Are there metrics that could be more robust?
- **Usability and Navigation:** Is there anything about the layout or navigation that could be more intuitive?
- **Troubleshooting Suggestions:** Are there any tips or contextual insights you’d find helpful when reviewing your performance metrics, especially for identifying and resolving issues faster?

We'll use both discussions (Enterprise preview back in July ) to collate your feedback 👍 

### Share your Wins 🏅 

We’d love to hear how you’re using the new Actions Performance Metrics  to make a difference in your workflows and job management. Has this feature helped you identify and address issues you previously couldn’t pinpoint? What changes did you make, and what was the outcome? Did the insights reveal any surprises in your workflows, such as hidden bottlenecks or frequent failure points? 

Share your stories and feedback in the comments below 👇👍

**Comment by giordano** [comment]
The dates in `/actions/metrics/performance` and `/actions/metrics/usage` are displayed in the weird format `month/day/year`, can we have the [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)-approved format `year-month-day`, or at very least `day/month/year`?  Thanks

***Edit***: it turns out my `LANG` env var was screwed up and was set to `C.UTF-8` despite I had previously set the system to use a different locale.  Resolving the issue also brought back a better display of the date on GitHub, which was clearly only using a system setting.  Sorry for the noise!

**Reply by samus-aran** [reply]
> weird format 

I appreciate this comment, it made me chuckle 😅

**Comment by JamesBurnside** [comment]
Is there a way to filter by branch? I'd love to be able to filter the insights of my team's action runs by our main branch to determine stability and performance in scenarios where it should pass 100%

**Reply by philip-gai** [reply]
There is not a way to filter by branch currently. But this is interesting feedback, thank you!

**Reply by austinpray-mixpanel** [reply]
We monitor the same thing via Mixpanel right now. In addition to ability to filter by branch you also need ability to filter by event (push, workflow_dispatch, schedule, etc). GitHub Actions triggered via push to our default branch are expected to pass 100% of the time. Branches triggered via workflow_dispatch or on a schedule should definitely pass but they have a different SLA for us. Like a cron job that deletes stale branches failing is lower priority than a test suite failing in the default branch (all developers are blocked).

![image](https://github.com/user-attachments/assets/2ec30e8a-4339-412e-a056-5ebb8b2fb145)

**Reply by mrmckeb** [reply]
We also need the ability to filter by branch, as we conditionally run/change steps depending on the branch and when in the merge queue.

**Comment by astendal** [comment]
This is a much-needed addition, thanks for your work on this. For an enterprise like ours with more than 100 organizations, this won't be very useful for administration until we can get an enterprise-level version of this dashboard. 

That said, here's some additional views we'd love to see here (the most important ones to our enterprise are at the top): 
- Summary (i.e, a fixed number) of current jobs in each state (queued, running, etc), filterable by runner group
- Graph of average and 95th percentile duration in each state, filterable by runner group
- Queue table per runner group
- Top N users and repositories by number of concurrent queued and running jobs (i.e, noisy neighbors)
- Execution duration by reusable component, e.g. "actions/checkout average run time: 15 seconds" 

*Some of this* ☝️ *is related to [the feedback I left on the usage metrics discussion](https://github.com/orgs/community/discussions/132817#discussioncomment-10329647).*

If you have API support in the pipeline sooner than an enterprise-level dashboard, we'd love to know so we can plan to cobble together our own enterprise-level summaries.

**Reply by philip-gai** [reply]
An enterprise-level dashboard and an API are both on our roadmap! Thanks for this great feedback! 😄

**Reply by satya2p** [reply]
@philip-gai  Do you have an ETA ?

**Reply by philip-gai** [reply]
No I do not, however we look at the backlog each quarter and this is high on the list!

**Comment by pavlospt** [comment]
Is there an API for this? We would love to have one to integrate those metrics in application related dashboards 🙏🏻

**Reply by philip-gai** [reply]
There is not an API for this yet but it's on our roadmap!

**Reply by pavlospt** [reply]
Looking forward to it, as we recently had to built that from scratch and we most certainly might be missing some edge cases that this dashboard will probably dont miss 😄

**Comment by andrewjmeier** [comment]
Is the average for run time mean or median? It would be great to be able to remove some outliers or break them down by certain percentiles. 

I think our data is skewed short by jobs that get cancelled early as well. Maybe those could be excluded?

**Reply by philip-gai** [reply]
The average for the run time is the mean. Your request makes sense - thank you for the feedback! 📓

**Comment by JorenVsyntra** [comment]
i dont understand..

**Comment by lure8225** [comment]
Any idea in which time-frame this makes it to the GHES (on premise) if at all? 
We recently started migrating more things from a Jenkins environment and the lack of this and similar statstics/metrics seems to be one of the biggest pain points.

**Reply by lkfortuna** [reply]
Hi there 👋  We currently do not have plans to bring this feature to GHES. The biggest reason is that we do not have access to the metrics on GHES instances since they are on private servers. There would be a need for you to send the metrics to us first, and then we would resurface them back to you.

**Reply by lure8225** [reply]
Thanks for the information but very sad to hear this. My expectation/hope was the data collection and all the backend would be integrated in the GHES server.

**Comment by arnomijs** [comment]
Hi,

I have a couple of questions/remarks:

- Can the aggregations on top change based on the filters selected? This is not the case at the moment.

- Can custom properties be introduced as a filter option? This would be really helpful filtering out repo's.

- Can the reusable workflow be added as a filter option instead of to the calling workflow name?

- Will this be able to serve as a real time monitoring solution in the future? I see that now it's mostly to 21 minutes ago, 1 hour ago, ...
Are there plans to decrease this interval to be (almost) real time?

Thanks in advance, great feature!

**Comment by Rymania** [comment]
I am interested at looking at the 'called' workflow usage so that we can measure reuse of our reusable actions and workflows. Right now I cannot see a way of doing this; the caller workflows and actions are displayed.

Example: a ci.yml workflow in a given repository calls:

- reuse-build.yml
- reuse-deploy.yml
- reuse-test.yml

As the owner of the 3 called workflows I am interested in usage of them, however they are hidden in the GitHub Actions Org Insights because they only show the usage of ci.yml for that given run.

In addition some of these are stacked a few levels deep in the hierarchy (e.g. reuse-build may have reuse-action-1 as part of it) - so the API isn't much use either (as detailed here  https://docs.github.com/en/actions/sharing-automations/reusing-workflows#monitoring-which-workflows-are-being-used)

**Reply by astendal** [reply]
I would LOVE to be able to measure performance of reusable actions and workflows. +1 to this.

**Comment by nmiddendorff** [comment]
Any plans to add additional visualizations to the new pages? It would be great to see a time series that shows how failure rate and execution time are trending over time.

**Comment by acasademont** [comment]
Hi! We would love to have as @andrewjmeier pointed out some more detailed stats.

Also, in our case, failed builds normally take much less time (they fail early). So in a type of job where the % failure rate is high the average time is normally lower. We would love to measure the run times only for **successful** runs, the cancelled/failed ones distort the metrics too much. Thanks a lot!

**Reply by acasademont** [reply]
being more specific about the "detailed stats". We're very interested in the "queue time" metrics. Seeing the average is great but for finding outliers is not that useful. We suspect that at certain points of the day the "queue time" is huge and in others is almost 0. would be great to see the p50 (median), p75, p90, p95, p99 and max stats for that metric along with the average.

thank you!

**Comment by server-mm404** [comment]
> We are excited to introduce a new observability UI that provides insights into the performance of your workflows and jobs for your organizations or repositories, now available in public preview! Performance metrics are designed to help you monitor, analyze, and optimize your Actions workflows and job execution. With these insights, you can answer common questions like:
> 
> * **How long does it take for my workflows or jobs to complete?** Gain visibility into the end-to-end execution times of your workflows and jobs, helping you identify bottlenecks and areas for improvement.
> * **How long are my workflows or jobs waiting to run?** View queue times to identify bottlenecks in job scheduling, enabling you to make adjustments that reduce idle time and improve throughput.
> * **Which of my workflows or jobs are consistently failing?** Spot frequent failures in your workflows, pinpoint root causes, and take preemptive actions to improve reliability and performance.
> * **Where are my longest-running workflows or jobs originating from?**
>   Get insights on job origin points, allowing you to address lengthy processes directly and refine areas with the most significant impact.
> 
> ![alt_text](https://github.com/user-attachments/assets/a5a5523f-28d3-4415-8ae9-bc9653f29dc1)
> 
> ### Getting Started ➡️
> To start leveraging these insights, navigate to your organization or repository homepage, select **“Insights” **near the top of the page, and then choose **“Actions Performance Metrics”** on the left side of the page. Once you’re there, you’ll have public preview access to detailed metrics on your workflows and jobs, offering a comprehensive view of performance at every stage.
> 
> ### GitHub Actions Metrics for Free, Pro, and Team plans 🆓
> We are also pleased to announce that with today's release, GitHub Actions Metrics are now available to Free, Pro, and Team plans. Previously, this feature was [only available to those on the GitHub Enterprise Cloud plan](https://github.com/orgs/community/discussions/132817).
> 
> To learn more about GitHub Actions Metrics, check out our [public documentation](https://docs.github.com/en/actions/administering-github-actions/viewing-github-actions-metrics)
> 
> ### Share your Feedback 💬
> We’re committed to making Actions Performance metrics as helpful and actionable as possible, and hearing directly from you will guide future improvements. As you explore the new feature, keep these questions in mind to help guide your experience:
> 
> * **Metrics or Data Points:** Are there additional metrics or data points you’d like to see included? Are there metrics that could be more robust?
> * **Usability and Navigation:** Is there anything about the layout or navigation that could be more intuitive?
> * **Troubleshooting Suggestions:** Are there any tips or contextual insights you’d find helpful when reviewing your performance metrics, especially for identifying and resolving issues faster?
> 
> We'll use both discussions (Enterprise preview back in July ) to collate your feedback 👍
> 
> ### Share your Wins 🏅
> We’d love to hear how you’re using the new Actions Performance Metrics to make a difference in your workflows and job management. Has this feature helped you identify and address issues you previously couldn’t pinpoint? What changes did you make, and what was the outcome? Did the insights reveal any surprises in your workflows, such as hidden bottlenecks or frequent failure points?
> 
> Share your stories and feedback in the comments below 👇👍

**Comment by server-mm404** [comment]
> We are excited to introduce a new observability UI that provides insights into the performance of your workflows and jobs for your organizations or repositories, now available in public preview! Performance metrics are designed to help you monitor, analyze, and optimize your Actions workflows and job execution. With these insights, you can answer common questions like:
> 
> * **How long does it take for my workflows or jobs to complete?** Gain visibility into the end-to-end execution times of your workflows and jobs, helping you identify bottlenecks and areas for improvement.
> * **How long are my workflows or jobs waiting to run?** View queue times to identify bottlenecks in job scheduling, enabling you to make adjustments that reduce idle time and improve throughput.
> * **Which of my workflows or jobs are consistently failing?** Spot frequent failures in your workflows, pinpoint root causes, and take preemptive actions to improve reliability and performance.
> * **Where are my longest-running workflows or jobs originating from?**
>   Get insights on job origin points, allowing you to address lengthy processes directly and refine areas with the most significant impact.
> 
> ![alt_text](https://github.com/user-attachments/assets/a5a5523f-28d3-4415-8ae9-bc9653f29dc1)
> 
> ### Getting Started ➡️
> To start leveraging these insights, navigate to your organization or repository homepage, select **“Insights” **near the top of the page, and then choose **“Actions Performance Metrics”** on the left side of the page. Once you’re there, you’ll have public preview access to detailed metrics on your workflows and jobs, offering a comprehensive view of performance at every stage.
> 
> ### GitHub Actions Metrics for Free, Pro, and Team plans 🆓
> We are also pleased to announce that with today's release, GitHub Actions Metrics are now available to Free, Pro, and Team plans. Previously, this feature was [only available to those on the GitHub Enterprise Cloud plan](https://github.com/orgs/community/discussions/132817).
> 
> To learn more about GitHub Actions Metrics, check out our [public documentation](https://docs.github.com/en/actions/administering-github-actions/viewing-github-actions-metrics)
> 
> ### Share your Feedback 💬
> We’re committed to making Actions Performance metrics as helpful and actionable as possible, and hearing directly from you will guide future improvements. As you explore the new feature, keep these questions in mind to help guide your experience:
> 
> * **Metrics or Data Points:** Are there additional metrics or data points you’d like to see included? Are there metrics that could be more robust?
> * **Usability and Navigation:** Is there anything about the layout or navigation that could be more intuitive?
> * **Troubleshooting Suggestions:** Are there any tips or contextual insights you’d find helpful when reviewing your performance metrics, especially for identifying and resolving issues faster?
> 
> We'll use both discussions (Enterprise preview back in July ) to collate your feedback 👍
> 
> ### Share your Wins 🏅
> We’d love to hear how you’re using the new Actions Performance Metrics to make a difference in your workflows and job management. Has this feature helped you identify and address issues you previously couldn’t pinpoint? What changes did you make, and what was the outcome? Did the insights reveal any surprises in your workflows, such as hidden bottlenecks or frequent failure points?
> 
> Share your stories and feedback in the comments below 👇👍

**Comment by pulkitanz** [comment]
- I would love to see usage metrics of each runner group, as in how many runners have been idle for how long so that we can reduce the cost by reducing machines, currently we do this by sending runner statuses to our databases and reviewing it but some sort of capability to check this in the Actions Performance Metrics would help.
- API for Actions Performance Metrics would be very useful.
- The filter does not work for multiple runner labels. 
- A filter to only list the jobs which exist on default branch would help as well.

**Comment by mustafaakin** [comment]
Why do you deprecate Get Workflow Usage endpoints? https://github.blog/changelog/2025-02-02-actions-get-workflow-usage-and-get-workflow-run-usage-endpoints-closing-down/

Is there replacement API coming? The discussion link on the page is not working and I think this is the most related one. But I'd like to get per workflow usage and find some optimizations on my own through API, removing an existing API without any replacement does not make any sense
